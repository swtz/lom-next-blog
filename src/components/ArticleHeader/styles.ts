import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      font-size: ${theme.font.sizes.xxlarge};
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.lteMedium} {
        font-size: ${theme.font.sizes.large};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    margin: ${theme.spacings.medium} 0;

    @media ${theme.media.lteMedium} {
      font-size: 2rem;
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
