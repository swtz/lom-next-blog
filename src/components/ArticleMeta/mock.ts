import { ArticleMetaProps } from '.';

export default {
  createdAt: '2025-02-12T11:41:13.991Z', // value from current Strapi API
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'surfista',
        slug: 'surfista',
      },
    },
  },

  categories: {
    data: [
      {
        id: '4',
        attributes: {
          displayName: 'Off Topic',
          slug: 'off-topic',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'tech',
          slug: 'tech',
        },
      },
    ],
  },
} as ArticleMetaProps;
