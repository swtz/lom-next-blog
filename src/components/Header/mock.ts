import { HeaderProps } from '.';

export default {
  blogName: 'Blog do Arroz Ã© vida!',
  blogDescription: `Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa.`,
  logo: '/assets/images/logo.svg',
  showText: true,
} as HeaderProps;
