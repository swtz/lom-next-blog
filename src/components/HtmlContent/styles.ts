import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: calc(${theme.font.sizes.xsmall} + 0.2rem);
    line-height: 1.5;

    p {
      margin: ${theme.spacings.medium} 0;
    }

    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    code {
      font-family: monospace;
      color: ${theme.colors.secondary};
      font-size: ${theme.font.sizes.xsmall};
      background: ${theme.colors.mediumGray};
      padding: 0.2rem;
      margin: 0.2rem;
    }

    pre {
      background: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.medium} 0;
      font-size: ${theme.font.sizes.xsmall};
      width: 100%;
      overflow-x: auto;
    }

    pre code {
      color: inherit;
      background: inherit;
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }

    .image figcaption {
      background: ${theme.colors.mediumGray};
      font-size: ${theme.font.sizes.xsmall};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }

    .image-style-side {
      float: right;
      max-width: 50%;
      margin: ${theme.spacings.medium} ${theme.spacings.small};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }

    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    .media iframe {
      width: 100%;
      height: auto;
      aspect-ratio: 16 / 9;
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.darkerGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      pre {
        font-size: 1.2rem;
      }

      .image-style-side {
        float: none;
        max-width: 100%;
        margin: ${theme.spacings.xlarge} 0;
      }
    }
  `};
`;
