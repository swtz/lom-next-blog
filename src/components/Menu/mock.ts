import { MenuProps } from '.';

export default {
  text: 'Hello, LogoLink!',
  srcImg: 'assets/images/logo.svg',
  link: 'https://www.jw.org',
  newTab: false,
  links: [
    {
      id: 10,
      children: 'intro',
      link: '#intro',
      newTab: false,
    },
    {
      id: 8,
      children: 'grid-one',
      link: '#grid-one',
      newTab: false,
    },
    {
      id: 9,
      children: 'gallery',
      link: '#gallery',
      newTab: false,
    },
    {
      id: 11,
      children: 'grid-two',
      link: '#grid-two',
      newTab: false,
    },
    {
      id: 7,
      children: 'contact',
      link: '#contact',
      newTab: false,
    },
    {
      id: 12,
      children: 'Dev',
      link: 'https://github.com/swtz',
      newTab: true,
    },
  ],
} as MenuProps;
