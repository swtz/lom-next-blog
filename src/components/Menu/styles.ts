import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css``}
`;

export const Nav = styled.nav`
  ${() => css``}
`;

export const Logo = styled.div`
  ${() => css``}
`;

export const OpenClose = styled.a`
  ${() => css``}
`;
