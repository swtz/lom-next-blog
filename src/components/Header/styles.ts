import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    /* padding: ${theme.spacings.xhuge}; */
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.xsmall};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.xsmall} - 1rem);
    }

    @media ${theme.media.lteXSmall} {
      flex-flow: column wrap;

      > div {
        ${HeadingStyles} {
          margin: ${theme.spacings.xsmall} 0
            calc(${theme.spacings.small} - 1rem);
        }
      }
    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
