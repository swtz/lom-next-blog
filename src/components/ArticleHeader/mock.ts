import { ArticleHeaderProps } from '.';
import { ArticleMetaProps } from '../ArticleMeta';

const metadata = {
  createdAt: '2025-02-12T11:41:13.991Z',
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'surfista',
        slug: 'surfista',
      },
    },
  },

  categories: {
    data: [
      {
        id: '4',
        attributes: {
          displayName: 'Off Topic',
          slug: 'off-topic',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'tech',
          slug: 'tech',
        },
      },
    ],
  },
} as ArticleMetaProps;

export default {
  title: 'Hello, rice! With rice is possible make a several delicious dishes',
  excerpt: `Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa (Asian rice).`,
  cover: {
    data: {
      id: 1,
      attributes: {
        url: '/assets/images/arroz-branco-saudavel.webp',
        alternativeText: null,
      },
    },
  },
  metadata: metadata,
} as ArticleHeaderProps;
