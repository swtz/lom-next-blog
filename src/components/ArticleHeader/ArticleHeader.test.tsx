import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ArticleHeader } from '.';

import mock from './mock';

describe('<ArticleHeader />', () => {
  it('should render a title, a description, an image and the post metadata', () => {
    renderTheme(<ArticleHeader {...mock} />);

    expect(
      screen.getByRole('heading', { name: mock.title }),
    ).toBeInTheDocument();

    expect(screen.getByAltText(`${mock.excerpt}`)).toHaveAttribute(
      'alt',
      `${mock.excerpt}`,
    );

    expect(
      screen.getByRole('img', { name: `${mock.excerpt}` }),
    ).toHaveAttribute('src', '/assets/images/arroz-branco-saudavel.webp');

    expect(
      screen.getByRole('link', {
        name: `${mock.metadata.author.data.attributes.displayName}`,
      }),
    ).toBeInTheDocument();
  });

  it('should render a image with alternative text', () => {
    const image = {
      data: {
        id: 1,
        attributes: {
          url: '/assets/images/arroz-branco-saudavel.webp',
          alternativeText: 'arroz-no-prato-com-erva-em-cima',
        },
      },
    };

    renderTheme(<ArticleHeader {...mock} cover={image} />);

    expect(
      screen.getByRole('img', { name: 'arroz-no-prato-com-erva-em-cima' }),
    ).toHaveAttribute('alt', `${image.data.attributes.alternativeText}`);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<ArticleHeader {...mock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
