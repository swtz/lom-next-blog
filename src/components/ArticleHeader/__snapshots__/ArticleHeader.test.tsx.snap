// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleHeader /> should match snapshot 1`] = `
.c5 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c5 .categories span::after {
  content: ', ';
}

.c5 .categories span:last-child:after {
  content: '';
}

.c5 a {
  color: #DC143C;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c5 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c2 {
  color: #000000;
  line-height: 1.2;
  font-size: 6.4rem;
  text-transform: none;
}

.c0 {
  padding-bottom: 4.0rem;
  margin-bottom: 4.0rem;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c0 .c1 {
  font-size: 4.8rem;
  margin: 0;
  margin-bottom: 2.4rem;
}

.c3 {
  font-size: 2.4rem;
  margin: 2.4rem 0;
}

.c4 {
  max-width: 100%;
  display: block;
  margin-bottom: 2.4rem;
}

@media (max-width:768px) {
  .c2 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c0 .c1 {
    font-size: 3.2rem;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 2rem;
  }
}

<header
  class="c0"
>
  <h2
    class="c1 c2"
  >
    Hello, rice! With rice is possible make a several delicious dishes
  </h2>
  <p
    class="c3"
  >
    Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa (Asian rice).
  </p>
  <img
    alt="Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa (Asian rice)."
    class="c4"
    src="/assets/images/arroz-branco-saudavel.webp"
  />
  <div
    class="c5"
  >
    <p>
      <span>
        Por 
      </span>
      <a
        href="/author/surfista"
      >
        surfista
      </a>
      <span
        class="separator"
      >
         | 
      </span>
      <time
        datetime="2025-02-13T11:41:13.991Z"
      >
        13 de fev. de 2025
      </time>
      <span
        class="separator"
      >
         | 
      </span>
      <span
        class="categories"
      >
        <span>
          <a
            href="/category/off-topic"
          >
            Off Topic
          </a>
        </span>
        <span>
          <a
            href="/category/tech"
          >
            tech
          </a>
        </span>
      </span>
    </p>
  </div>
</header>
`;
